include/database/MysqliManager.php

```diff
@@ -348,13 +348,17 @@ public function connect(array $configOptions = null, $dieOnError = false)
         }
 
         // cn: using direct calls to prevent this from spamming the Logs
-
         $collation = $this->getOption('collation');
-        if (!empty($collation)) {
-            $names = "SET NAMES 'utf8' COLLATE '$collation'";
+        $charset = $this->getOption('charset');
+
+        if (!empty($collation) && !empty($charset)) {
+            $names = "SET NAMES '$charset' COLLATE '$collation'";
```

Comment:
Please use quote or quoted methods on variable before concatenate it into an sql query.